libvast/src/segment_store.cpp

```diff
@@ -379,9 +378,19 @@ void segment_store::inspect_status(caf::settings& xs,
 }
 
 caf::error segment_store::register_segments() {
-  for (auto filename : directory{segment_path()})
-    if (auto err = register_segment(filename))
-      return err;
+  auto p = std::filesystem::path{segment_path().str()};
+  if (!std::filesystem::exists(p))
+    return caf::none;
+  std::error_code err{};
+  std::filesystem::directory_iterator dir{p, err};
```

Comment:
(Just a side note.)

Letting TOCTTOU issues aside, I wonder if we can leverage some invariants along the lines of if (a) a path exists, and (b) it's a directory, then the `directory_iterator` construction will not fail. This would make iterating over directories a little easier because you could write:

```cpp
if (exists && is_directory)
  for (const auto& entry : std::filesystem::directory_iterator{p})
    if (entry.exists())
       ...
```